annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -385,8 +386,13 @@ private String resolveHgvspShortFromAAs() {
         try {
             String[] aaParts = canonicalTranscript.getAminoAcids().split("/");
             if (canonicalTranscript.getConsequenceTerms() != null && canonicalTranscript.getConsequenceTerms().get(0).equals("inframe_insertion")) {
-                hgvsp = aaParts[1].substring(0,1) + canonicalTranscript.getProteinStart() + "_" + aaParts[1].substring(1, 2) + "ins" +
-                        canonicalTranscript.getProteinEnd() + aaParts[1].substring(2);
+                if (canonicalTranscript.getHgvsc() != null && canonicalTranscript.getHgvsc().contains("dup")) {
+                    hgvsp = aaParts[1].substring(0,1) + String.valueOf(Integer.valueOf(canonicalTranscript.getProteinStart()) - 1 ) + "dup";
```

Comment:
I'm sure this is a larger issue (outside the scope of this PR) but I think that having string parsing of attributes in client code is a sign of fragility. If we need to be breaking up the getAminoAcids() result into subfields and taking substrings of those fields we probably should implement this logic in a model class or in a parser class. For one thing it would make reading / comprehending this code easier, since it is not obvious what is in the first character of the second field .. so a named method would be self-documenting.

Reply:
I think having some examples of Hgvsc and amino acid parts in comments would be helpful.